src/qibo/backends/clifford.py

```diff
@@ -589,8 +105,31 @@ def zero_state(self, nqubits: int):
         symplectic_matrix[nqubits:-1, nqubits : 2 * nqubits] = self.np.copy(I)
         return symplectic_matrix
 
+    def _clifford_pre_execution_reshape(self, state):
+        """Reshape the symplectic matrix to the shape needed by the engine before circuit execution.
+
+        Args:
+            state (ndarray): The input state.
+
+        Returns:
+            (ndarray): The reshaped state.
+        """
+        return self.engine._clifford_pre_execution_reshape(state)
+
+    def _clifford_post_execution_reshape(self, state, nqubits):
+        """Reshape the symplectic matrix to the shape needed by the engine after circuit execution.
+
+        Args:
+            state (ndarray): The input state.
+            nqubits (int): Number of qubits.
+
+        Returns:
+            (ndarray): The reshaped state.
+        """
+        return self.engine._clifford_post_execution_reshape(state, nqubits)
```

Comment:
In principle, it's just a function consisting of a single function call. You could simplify and inline the inner function call in the execution (these functions should be called in a single place, if I'm not wrong)

Reply:
Ok, the first one is actually used twice...